src/lib/utils/logger.ts

```diff
@@ -1,10 +1,5 @@
 import noop from './noop';
 
-type Deprecate<TCallback = (...args: any[]) => any> = (
```

Comment:
this previous form caused the generic to already be filled when deprecate gets declared, which is why it must be inline

Reply:
I don't get this, what do you mean?

Reply:
```
let deprecate: Deprecate = ...
```

deprecate is no longer generic there, and I didn't find a way to make a let generic (only a function, type or class)

Reply:
https://github.com/microsoft/TypeScript/issues/17574